src/ray/gcs/redis_context.cc

```diff
@@ -172,55 +152,71 @@ const std::vector<std::optional<std::string>> &CallbackReply::ReadAsStringArray(
   return string_array_reply_;
 }
 
-// This is a global redis callback which will be registered for every
-// asynchronous redis call. It dispatches the appropriate callback
-// that was registered with the RedisCallbackManager.
-void GlobalRedisCallback(void *c, void *r, void *privdata) {
-  if (r == nullptr) {
-    return;
+RedisRequestContext::RedisRequestContext(instrumented_io_context &io_service,
+                                         RedisCallback callback,
+                                         RedisAsyncContext *context,
+                                         std::vector<std::string> args)
+    : exp_back_off_(RayConfig::instance().redis_retry_base_ms(),
+                    RayConfig::instance().redis_retry_multiplier(),
+                    RayConfig::instance().redis_retry_max_ms()),
+      io_service_(io_service),
+      redis_context_(context),
+      pending_retries_(RayConfig::instance().num_redis_request_retries()),
```

Comment:
should we always initialize pending_retries_ to at least 1?

Reply:
yea, i think we should defined the num_redis_request_retries to be the retris we'll do if the first failed.